src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftAPIService.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.elasticsearch.plugin;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.ElasticsearchSecurityException;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.rest.RestStatus;
+
+import com.jayway.jsonpath.JsonPath;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class OpenshiftAPIService {
+    
+    private static final String ACCEPT = "Accept";
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final Logger LOGGER = Loggers.getLogger(OpenshiftAPIService.class);
+    
+    public String userName(final String token) {
+        Response response = null;
+        try (DefaultOpenShiftClient client = buildClient(token)) {
+            Request okRequest = new Request.Builder()
+                    .url(client.getMasterUrl() + "apis/user.openshift.io/v1/users/~")
```

Comment:
Does the `~` here mean "the user whose token this is"?

Reply:
Is there a way to request only the `metadata` or `metadata.name`?  Purely in the interest of reducing the response payload.

Reply:
Correct.  This means 'give me info about the user who is associated with this token'.  I know of no way to reduce the size of the payload.  The oc binary does the same; it gets the entire JSON but only presents you with a formatted output